protocol/src/main/java/io/camunda/zeebe/protocol/record/intent/Intent.java

```diff
@@ -109,15 +111,17 @@ static Intent fromProtocolValue(final ValueType valueType, final short intent) {
         return DecisionEvaluationIntent.from(intent);
       case CHECKPOINT:
         return CheckpointIntent.from(intent);
+      case PROCESS_INSTANCE_MODIFICATION:
+        return ProcessInstanceModificationIntent.from(intent);
       case NULL_VAL:
       case SBE_UNKNOWN:
         return Intent.UNKNOWN;
-      default:
-        throw new RuntimeException(
-            String.format(
-                "Expected to map value type %s to intent type, but did not recognize the value type",
-                valueType.name()));
     }
+
+    throw new RuntimeException(
+        String.format(
+            "Expected to map value type %s to intent type, but did not recognize the value type",
+            valueType.name()));
```

Comment:
🙃 Personal preference, but I like it better to have this as the default with the warning suppression. A switch expression would be even nicer, but Java8 😢